babel/dates.py

```diff
@@ -1215,8 +1214,16 @@ def __init__(self, pattern, format):
     def __repr__(self):
         return '<%s %r>' % (type(self).__name__, self.pattern)
 
-    def __unicode__(self):
-        return self.pattern
+    if PY2:
```

Comment:
To be honest, I'd be more comfortable if the PY2 code lived within the functions instead of the class definition changing depending on interpreter version:

```python
def __unicode__(self):
    return self.pattern

def __str__(self):
    pat = self.pattern
    if PY2:
        pat = pat.encode('utf-8')
    return pat
```